richtextfx/src/main/java/org/fxmisc/richtext/model/TextChange.java

```diff
@@ -5,30 +5,39 @@
 
 public abstract class TextChange<S, Self extends TextChange<S, Self>> {
 
-    public static enum ChangeType {
+    /**
+     * Indicates whether this change can be merged with another change if they are both {@link #INSERTION} or
+     * both {@link #DELETION}.
+     */
+    public static enum MergeType {
         /** Indicates that the change will insert something but not remove anything */
         INSERTION,
         /** Indicates that the change will delete something but not insert anything */
         DELETION,
-        /** Indicates that the change will remove something and insert something as its replacement.
-         * For {@link PlainTextChange}s, the replacement will only be content.
-         * For {@link RichTextChange}s, the replacement may only be the style, only the content, or a mix of both.
-         * Style changes may change the style of the paragraph's content or a non-empty {@link Paragraph}
-         * ({@code p.length() > 0}) */
-        REPLACEMENT,
-        /** Indicates that the paragraph style of an empty {@link Paragraph} ({@code p.length() == 0}) was changed */
-        RESTYLED_PARAGRAPH
+        /** Indicates that the change is a style change, a replacement or something other than the other two types */
+        NEITHER,
```

Comment:
I like `NONE` better, but it still leaves the similar question of "if it isn't none, then what is it?" :smile:

I think we'll just take the minor performance hit and make the code more readable by writing it like so:
````java
if (this.getType == INSERTION || this.getType() == DELETION)
    && this.getType() == that.getType() // and so forth
````